osquery/tables/system/linux/kernel_keys.cpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+#include <fstream>
+
+#include <boost/algorithm/string/trim.hpp>
+
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/split.h>
+
+namespace osquery {
+namespace tables {
+
+static const std::string kProcKeysPath = "/proc/keys";
+
+QueryData genKernelKeys(QueryContext& context) {
+  if (!pathExists(kProcKeysPath).ok()) {
+    VLOG(1) << "Cannot find keyrings file: " << kProcKeysPath;
+    return {};
+  }
+
+  std::string content;
+  auto res = osquery::readFile(kProcKeysPath, content);
+  if (!res.ok()) {
+    VLOG(1) << "Error reading keyrings file: " << kProcKeysPath;
+    return {};
+  }
+
+  QueryData results;
+  for (const auto& key : osquery::split(content, "\n")) {
+    Row r = {};
+    auto details = osquery::split(key, " ");
+    if (details.size() != 10) {
+      // Interesting error case, the key line is not well formed.
```

Comment:
Does this merit a verbose warning?

Reply:
Indeed, added a verbose warning instead of this comment.